pkg/api/types.go

```diff
@@ -451,6 +451,8 @@ type ISCSIVolumeSource struct {
 	IQN string `json:"iqn,omitempty"`
 	// Required: iSCSI target lun number
 	Lun int `json:"lun,omitempty"`
+	// Optional: Defaults to 'default' (tcp). iSCSI interface name that uses an iSCSI transport.
+	ISCSIInterface string `json:"iscsiInterface,omitempty"`
```

Comment:
See https://github.com/kubernetes/kubernetes/pull/16877#issuecomment-155241387 

This iscsi Interface often gets confused with network interface, and while they are named similarly, they are very different entities. Providing a parameter like "eth0" for Interface will cause the iscsi volume plugin to fail. Part of the reasoning behind the naming scheme is to ensure there is no ambiguity here. The default name for this file (iface.iscsi_ifacename) when generated by iscsiadm would be something like 
<transport_name.hwaddress>, eg cxgb4i.00:07:48:b3:e4:22 which might use eth0 interface, but the iscsi interface name would nevertheless be cxgb4i.00:07:48:b3:e4:22,not eth0.